connectors/aws/src/main/java/io/camunda/connector/aws/dynamodb/operation/item/UpdateItemOperation.java

```diff
@@ -0,0 +1,90 @@
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.aws.dynamodb.operation.item;
+
+import com.amazonaws.services.dynamodbv2.document.AttributeUpdate;
+import com.amazonaws.services.dynamodbv2.document.DynamoDB;
+import com.amazonaws.services.dynamodbv2.document.PrimaryKey;
+import com.amazonaws.services.dynamodbv2.model.AttributeAction;
+import com.google.gson.Gson;
+import io.camunda.connector.aws.GsonComponentSupplier;
+import io.camunda.connector.aws.dynamodb.model.item.UpdateItem;
+import io.camunda.connector.aws.dynamodb.operation.AwsDynamoDbOperation;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+
+public class UpdateItemOperation implements AwsDynamoDbOperation {
+  private final UpdateItem updateItemModel;
+
+  public UpdateItemOperation(final UpdateItem updateItemModel) {
+    this.updateItemModel = updateItemModel;
+  }
+
+  @Override
+  public Object invoke(final DynamoDB dynamoDB) throws InterruptedException {
+
+    PrimaryKey primaryKey = new PrimaryKey();
+    Gson gson = GsonComponentSupplier.gsonInstance();
+    gson.toJsonTree(updateItemModel.getPrimaryKeyComponents())
+        .getAsJsonObject()
+        .entrySet()
+        .forEach(
+            entry -> {
+              Object o =
+                  Optional.ofNullable(entry.getValue())
+                      .map(obj -> gson.fromJson(obj, Object.class))
+                      .orElse(null);
+              primaryKey.addComponent(entry.getKey(), o);
+            });
+
+    List<AttributeUpdate> attributeUpdates = new ArrayList<>();
+
+    gson.toJsonTree(updateItemModel.getKeyAttributes())
+        .getAsJsonObject()
+        .entrySet()
+        .forEach(
+            entry -> {
+              Object o =
+                  Optional.ofNullable(entry.getValue())
+                      .map(obj -> gson.fromJson(obj, Object.class))
+                      .orElse(null);
+              AttributeUpdate attributeUpdate;
+              AttributeAction attributeAction =
+                  AttributeAction.fromValue(
+                      updateItemModel.getAttributeAction().toUpperCase(Locale.ROOT));
+
+              switch (attributeAction) {
+                case PUT:
+                  attributeUpdate = new AttributeUpdate(entry.getKey()).put(o);
+                  break;
+                case DELETE:
```

Comment:
(Optional) Since we have java 17 you can try a new switch statement

```
attributeUpdate = switch(attributeAction) {
  case PUT: yield  new AttributeUpdate(entry.getKey()).put(o);
}
```